src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftImageTagger.java

```diff
@@ -132,113 +122,170 @@ default String deriveImageTag(String imageID, IImageStream srcIS) {
-    				newOverrides.remove("namespace");
-    				OpenShiftCreator isCreator = new OpenShiftCreator(getApiURL(newOverrides), destinationNS, getDestinationAuthToken(overrides), getVerbose(newOverrides), createJson);
-    				isCreator.setAuth(Auth.createInstance(chatty ? listener : null, getApiURL(newOverrides), overrides));
-    		    	//isCreator.setToken(getDestinationToken());
-    				
-    				
-    				boolean newISCreated = isCreator.coreLogic(launcher, listener, newOverrides);
-
-    				if (!newISCreated) {
-    					listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_CREATE_DEST_IS, getProdStream(overrides), destinationNS));
-    					return false;
-    				}
-    				
-    				destIS = client.get(ResourceKind.IMAGE_STREAM, getProdStream(overrides), destinationNS);
-    			}
-        	}
-        	
-        	if (destIS == null) {
-    			listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_GET_IS, getProdStream(overrides), destinationNS));
-    			return false;
-        	}
-        	
-        	// tag image
-        	destIS.addTag(getProdTag(overrides), tagType, srcImageID, getNamespace(overrides));
-			if (chatty)
-				listener.getLogger().println("\n updated image stream json " + ((ImageStream)destIS).getNode().toJSONString(false));
-			client.update(destIS);
-			
-			
+
+			List<String> newDestinationStreams = Arrays.asList( destStreams.split( "," ) );
+			List<String> newDestTags = Arrays.asList( destTags.split( "," ) );
+
+			/**
+			 * The following logic is permissive of the following combinations
+			 *  (1) 1 dest stream and 1 dest tag  	(i.e. a single new stream:tag combination)
+			 * 	(2) 1 dest stream and N dest tags   (i.e. all tags apply to the same stream)
+			 * 	(3) 1 dest tag and N dest streams	(i.e. all streams will get the same tag)
+			 * 	(4) N dest streams and N dest tags	(i.e. each index will be a unique stream:tag)
+			 */
+
+			if ( newDestinationStreams.size() != 1 && newDestTags.size() != 1 && newDestinationStreams.size() != newDestTags.size() ) {
+				String error = String.format( "Destination streams (%s) cardinality [%d] is incompatible with destination tag (%s) cardinality [%d]", newDestinationStreams, newDestinationStreams.size(), newDestTags, newDestTags.size() );
+				throw new IllegalArgumentException( error );
+			}
+
+			Iterator<String> iDestStreams = newDestinationStreams.iterator();
+			Iterator<String> iDestTags = newDestTags.iterator();
+			String destStream = null;
+			String destTag = null;
+			while ( iDestStreams.hasNext() || iDestTags.hasNext() ) {
+
+				if ( iDestStreams.hasNext() ) {
+					destStream = iDestStreams.next().trim();
+				}
+
+				if ( iDestTags.hasNext() ) {
+					destTag = iDestTags.next().trim();
+				}
+
+				IImageStream destIS = null;
+				for ( int retries = 10; retries > 0; retries-- ) {
```

Comment:
Precisely that. Modifying the resource repeatedly caused a 403 unless I pulled the resource again.